models/issue.go

```diff
@@ -1108,9 +1103,24 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 		}
 	}
 
-	// Milestone and assignee validation should happen before insert actual object.
-	if _, err = e.Insert(opts.Issue); err != nil {
-		return err
+	// There's no good way to identify a duplicate key error in database/sql; brute force some retries
+	dupIndexAttempts := issueMaxDupIndexAttempts
+	for {
```

Comment:
If we go back this way. We should define a max number of tries and trigger en error if we doesn't find a good number.